### PR TITLE
feat(boards): #MAG-5/6 (back) display+create all boards

### PR DESCRIPTION
**Pull request by flmartineau**
## Describe your changes

- Added 2 API endpoints for creating a board and fetching all boards

## Checklist tests

- check if both APIs are behaving properly with the expected parameters (see : https://confluence.support-ent.fr/display/RN/Boards)

## Issue ticket number and link

https://jira.support-ent.fr/browse/MAG-5
https://jira.support-ent.fr/browse/MAG-6

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [x] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/cgi/magneto/controller/BoardController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Post("/board")
+    @ApiDoc("Create a board")
+    @SecuredAction(Magneto.MANAGE_BOARD)
+    public void create(HttpServerRequest request) {
+        RequestUtils.bodyToJson(request, pathPrefix + "board", board -> {
+            UserUtils.getUserInfos(eb, request, user ->
+                    boardService.create(user, board)
+                        .onFailure(err -> {
+                                    log.error(err.getMessage());
```
Comment (lapeyronnie) [inline src/main/java/fr/cgi/magneto/controller/BoardController.java:55]: tu pourrais ajouter la classe dans le message d'erreur comme tu l'as fait pour laclasse DataHelper.

File: src/test/java/fr/cgi/magneto/model/BoardTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+    @Test
+    public void testBoardHasBeenInstantiated(TestContext ctx) {
+        Board board = new Board(boardJsonObject_1);
+        ctx.assertEquals(boardJsonObject_1, board.toJson());
+    }
+
+    @Test
+    public void testCourseHasContentWithObject(TestContext ctx) {
```
Comment (JinDouang) [inline src/test/java/fr/cgi/magneto/model/BoardTest.java:36]: "course"

File: src/test/java/fr/cgi/magneto/model/MongoQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Test
+    public void testMongoQueryHasBeenInstanciated(TestContext ctx) {
+        MongoQuery mongoQuery = new MongoQuery("collection")
+                .match(new JsonObject().put("field", "value"))
+                .matchRegex("regex", Arrays.asList("field1", "field2"))
+                .matchOr(new JsonArray().add(new JsonObject().put("field", "value")))
+                .sort("field", 1)
+                .project(new JsonObject().put("field", 1));
+        ctx.assertEquals(mongoQueryObject_1, mongoQuery.getAggregate());
```
Comment (JinDouang) [inline src/test/java/fr/cgi/magneto/model/MongoQueryTest.java:51]: pour compare ton json et ce qu'on s'attend vraiment ici!

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+        return promise.future();
+    }
+
+    private JsonObject getAllBoardsQuery(UserInfos user, Integer page,
+                                           String searchText, String folderId,
+                                           boolean isPublic,
+                                           boolean isShared, String sortBy, boolean getCount) {
+
+        //TODO: fetch shared boards (MAG-16)
```
[the remark this answers is not in the document]
Comment (flmartineau) [reply src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java:113]: oui je le précise en commentaire car pour l'instant le paramètre `sortBy` n'est pas utilisé

**Review by JinDouang — APPROVED** [review]

File: src/main/java/fr/cgi/magneto/controller/BoardController.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            boolean isPublic = Boolean.parseBoolean(request.getParam(Field.ISPUBLIC));
+            boolean isShared = Boolean.parseBoolean(request.getParam(Field.ISSHARED));
+            String sortBy = request.getParam(Field.SORTBY);
+            Integer page = request.getParam(Field.PAGE) != null ? Integer.parseInt(request.getParam(Field.PAGE)) : null;
+            boardService.getAllBoards(user, page, searchText, folderId, isPublic, isShared, sortBy)
+                    .onSuccess(result -> renderJson(request, result))
+                    .onFailure(fail -> {
+                        String message = String.format("[Magneto@%s::getAllBoards] Failed to get all boards : %s",
+                                this.getClass().getSimpleName(), fail.getMessage());
+                        log.error(message);
```
Comment (qpic-cgi) [inline src/main/java/fr/cgi/magneto/controller/BoardController.java:41]: tu gères déjà ton erreur dans ta méthode, tu ne risques pas d'avoir un spam de logs ?

File: src/main/java/fr/cgi/magneto/controller/BoardController.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    @ApiDoc("Create a board")
+    @SecuredAction(Rights.MANAGE_BOARD)
+    public void create(HttpServerRequest request) {
+        RequestUtils.bodyToJson(request, pathPrefix + "board", board ->
+                UserUtils.getUserInfos(eb, request, user ->
+                    boardService.create(user, board)
+                        .onFailure(err -> {
+                                    String message = String.format("[Magneto@%s::create] Failed to create board : %s",
+                                            this.getClass().getSimpleName(), err.getMessage());
+                                    log.error(message);
```
Comment (qpic-cgi) [inline src/main/java/fr/cgi/magneto/controller/BoardController.java:57]: cf https://github.com/CGI-OPEN-ENT-NG/magneto/pull/1/files#r970592943

File: src/main/java/fr/cgi/magneto/model/MongoQuery.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
+package fr.cgi.magneto.model;
+
+import io.vertx.core.json.*;
+
+import java.util.*;
+
+import static fr.cgi.magneto.Magneto.PAGE_SIZE;
+
+public class MongoQuery {
```
Comment (qpic-cgi) [inline src/main/java/fr/cgi/magneto/model/MongoQuery.java:9]: Dans statistics-presences on utilise l'object AggregationsBuilder pour construire les requêtes mongo, c'est relativement pratique !
https://github.com/OPEN-ENT-NG/presences/blob/1601004f41e7bacf57874175075d6139e81d3e77/statistics-presences/src/main/java/fr/openent/statistics_presences/bean/weekly/WeeklySearch.java#L38

Comment (qpic-cgi) [inline src/main/java/fr/cgi/magneto/model/MongoQuery.java:9]: Dans statistics-presences on utilise l'object AggregationsBuilder pour construire les requêtes mongo, c'est relativement pratique !
https://github.com/OPEN-ENT-NG/presences/blob/1601004f41e7bacf57874175075d6139e81d3e77/statistics-presences/src/main/java/fr/openent/statistics_presences/bean/weekly/WeeklySearch.java#L38

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

**Pull request merged**